server/api/cvrs.py

```diff
@@ -408,6 +421,274 @@ def parse_cvr_rows() -> Iterable[CvrBallot]:
+                    " The Tabulator and Batch fields in the CVR file"
+                    " must match the Tabulator and Batch Name fields in the"
+                    " ballot manifest."
+                    + (
+                        (
+                            " The closest match we found in the ballot manifest was:"
+                            f" {closest_match[0]}, {closest_match[1]}."
+                        )
+                        if closest_match
+                        else ""
+                    )
+                    + " Please check your CVR file and ballot manifest thoroughly"
+                    " to make sure these values match - there may be a similar"
+                    " inconsistency in other rows in the CVR file."
+                )
+
+    def parse_contest_metadata(cvr_csv: CSVIterator) -> CVR_CONTESTS_METADATA:
+        headers = next(cvr_csv)
+        # Based on files we've seen, the first 3 columns are Cast Vote Record,
+        # Precinct, Ballot Style and the rest are contest names
+        first_contest_column = 3
+        contest_names = headers[first_contest_column:]
+        contest_choices: Dict[str, Set[str]] = {
+            contest_name: set() for contest_name in contest_names
+        }
+
+        header_indices = get_header_indices(headers)
+
+        for row_index, row in enumerate(cvr_csv):
+            for contest_name in contest_names:
+                choice_name = column_value(
+                    row, contest_name, row_index + 1, header_indices, required=False
+                )
+                if choice_name:
+                    contest_choices[contest_name].add(choice_name)
+
+        # Assign each choice a column index in the interpretation string
+        contest_choice_pairs = [
+            (contest_name, contest_choice)
+            for contest_name, choices in contest_choices.items()
+            for contest_choice in sorted(choices)
+        ]
+        contest_choice_columns = {
+            choice: column for column, choice in enumerate(contest_choice_pairs)
+        }
+
+        return {
+            contest_name: dict(
+                # Until we know how vote-for-n contests are serialized in the
+                # ES&S CVR, we assume vote-for-1
+                votes_allowed=1,
+                choices={
+                    choice: dict(
+                        column=contest_choice_columns[(contest_name, choice)],
+                        num_votes=0,  # Will be counted while parsing rows
+                    )
+                    for choice in sorted(choices)
+                },
+                total_ballots_cast=0,  # Will be counted while parsing rows
+            )
+            for contest_name, choices in contest_choices.items()
+            if len(choices) > 0
+        }
+
+    def parse_interpretations(
+        cvr_csv: CSVIterator, contests_metadata: CVR_CONTESTS_METADATA
+    ) -> Iterable[Tuple[str, str]]:  # (CVR number, interpretations)
```

Comment:
In TypeScript I sometimes find it useful to alias common types like `string` with a more specific name. Perhaps you could do that here and ditch the comment.

Reply:
Good idea. I think in this case since it's just one spot, it's not worth aliasing, but I like the idea in general and can see it being useful.